### ISSUE TITLE
pthreads and neo4j-php-client

### ISSUE THREAD

**Issue by 0xfff** [krakjoe/pthreads#680]
In order to perform some work in the given timeframe i need to use a multi-threaded approach.
I am having issues with executing neo4j queries, any call to `php-client-object->run` yields a segfault

This is the run function from my thread class
```
public function run() {
        $id = $this->thread_id;
        $addr_array = $this->addr_array;
        $neo_ip = $this->neo_ip;
        $neo_timeout = $this->neo_timeout;
        $neo4j  = ClientBuilder::create()
        -> addConnection('default', $neo_ip)
        -> setDefaultTimeout($neo_timeout)
        -> build();

        $query = "MATCH a...."
            try{
                $result = $neo4j->run ($query);
                } 
            catch (Exception $e) {
            }
}
```
i get a segfault here: `$result = $neo4j->run ($query);`

any pointers would be appreciated.

**Comment by 0xfff** [comment]
some more info:
 if i call my `run` method directly, the script finishes as expected, but if i call `->start` i get a segfault.

I was assuming a share-nothing model between the threads - is that assumption incorrect?

**Comment by krakjoe** [comment]
As far as I can tell, this was fixed when we prepared for 7.2 ... if I'm wrong, please re-open the issue.
